### ISSUE TITLE
Upcoming boost 1.63 breaking change for template test cases

### ISSUE THREAD

**Issue by raffienficiaud** [etas/vs-boost-unit-test-adapter#165]
Hi there,

I do not know how you would be affected, but the template test cases will slighty change in 1.63. The rationale behind is https://svn.boost.org/trac/boost/ticket/12531 , and colons will get replaced with underscores.

**Comment by netspiri** [comment]
Hi @raffienficiaud and many thanks for letting us know early of this situation.

From our perspective, as long as the test name is consistent across `--list_content(=DOT)` and executing the same test using `--run_test`, we *should* be fine. If there are inconsistencies between the two, then we need to manually convert colons `:` to underscores `_` on our end as well.

Just a heads up, a similar issue may occur for the `--logger` argument when using file output sinks using fully rooted paths on Windows e.g. `C:\test.exe.log`.

Should you require further assistance or reporting on this, please let us know - glad to help.

**Comment by raffienficiaud** [comment]
Hi,

I created a fix for the issue you are mentioning (names with ':' like `C:\sthg.log`). Would be cool if you can test the corresponding branch `bug/log-sink-token-can-use-colon` because right now I do not have direct access to a Windows machine.

**Comment by netspiri** [comment]
Did a quick test from the command line with the following:
```sh
test.exe "--run_test=test_case_a" "--logger=XML,test_suite,C:\test.log.xml"
```
... and it failed with the following message:
```
Test setup error: Unable to determine the logger type from 'XML,test_suite,\Misc.test.boostd.exe.15160.6.test.log.xml'. Possible choices are: HRF, CLF, XML, JUNIT
```
Did another test with:
```sh
test.exe "--run_test=test_case_a" "--logger=XML,test_suite,\test.log.xml"
```
... and it failed as well with the same message.

Tried to slightly debug through it and it seems to stem from the comparison (specifically, the deference of the token iterator) in the following [line](https://github.com/boostorg/test/commit/18fc2b0b19f33f9d911bc0f4221bb35a75b7e987#diff-f0191b59e0a931218c6e0b8423234a22L910).

Should you require further assistance, please let us know.

**Comment by netspiri** [comment]
@guwirth I think this issue can be safely closed as discussed in #168.
